design/keycloak.x/configuration.md

```diff
@@ -620,6 +620,98 @@ keep it up to date with the latest release.
 For SPIs and Providers we will extend the ProviderFactory to add an option method to return information on what
 properties it supports. This will allow including SPI/provider specific configuration in the documentation.
 
+## Troubleshooting
+
+Keycloak should provide the appropriate leve of information and also tools for troubleshooting. It should help users to:
+
+* Identity the problem through clear and meaningful messages
+* Report issues by providing information about the error, server configuration, and environment.
+* Fix issues by providing suggestions on how to fix well-known usage mistakes and issues
+
+### Using the `version` command
+
+The `version` command should print out information about the runtime environment such as:
+
+* Keycloak Version
+* Operational System Info
+* JVM Info
+
+An example of output is:
+
+```
+Keycloak 13.0.0-SNAPSHOT
+JVM: 11.0.1 (Oracle Corporation OpenJDK 64-Bit Server VM 11.0.1+13)
+OS: Linux 5.10.15-200.fc33.x86_64 amd64
+```
+
+### Using the `show-config` command
+
+The `show-config` command should print out information about the server configuration. It should include:
+
+* Current profile
+* Runtime properties
+  * For each property indicate their source (persisted, CLI, system property, environment vars, etc)
+* Features enabled
+* List of installed providers
+  * Default providers
+  * Configuration for each provider
+* Quarkus properties
+
+An example of output is:
+
+```
+Current Profile: none
+Runtime Properties:
+        kc.cluster =  local (PersistedConfigSource)
```

Comment:
From the usability perspective, it can be also useful if at the end of "show-config" page, there is some glossary with the explanation what means `PersistedConfigSource`, what means `SysPropConfigSource` etc.

Reply:
Or perhaps change the out put to show `from server image` and `from system properties` ?

Reply:
Yes, that is good point. Question is, if it would be 100% clear for typical administrator what exactly it means? But maybe we can have both? Like the nicer names (`from system properties`) and also have the glossary at the end of the page? Not sure what is best for usability TBH...